src/containers/BuyButton/index.tsx

```diff
@@ -56,16 +57,31 @@ const BuyButton = withSuspense(
         d: erc721_id,
         e: premiumUint256,
       }
+      const balanceResult = await callContract(
+        ERC20_CONTRACT_INSTANCE,
+        'balanceOf',
+        address
+      )
+
+      const existingBalance = uint256.uint256ToBN(balanceResult[0])
```

Comment:
I don't like these 0 indexes either but in light of a very tight deadline, will come back to it later